### PR TITLE
Fix storage volume export for cephfs

### PR DESCRIPTION
**Pull request by monstermunchkin**

### CODE REVIEW

**Comment by monstermunchkin** [comment]
I'm converting this to draft as I'm unsure if the storage import works correctly for cephfs due to the symlinks.

File: lxd/storage/drivers/generic_vfs.go [hunk 1]
```diff
@@ -556,6 +556,18 @@ func genericVFSBackupVolume(d Driver, vol Volume, tarWriter *instancewriter.Inst
 				}
 
 				d.Logger().Debug(logMsg, log.Ctx{"sourcePath": mountPath, "prefix": prefix})
+
+				// Pass the target if mountPath is a symlink.
```
Comment (tomponline) [inline lxd/storage/drivers/generic_vfs.go:560]: "Follow the target"?

Comment (monstermunchkin) [reply lxd/storage/drivers/generic_vfs.go]: Fixed

File: lxd/storage/drivers/generic_vfs.go [hunk 2]
```diff
@@ -556,6 +556,18 @@ func genericVFSBackupVolume(d Driver, vol Volume, tarWriter *instancewriter.Inst
 				}
 
 				d.Logger().Debug(logMsg, log.Ctx{"sourcePath": mountPath, "prefix": prefix})
+
+				// Follow the target if mountPath is a symlink.
+				// Functions like filepath.Walk() won't list any directory content otherwise.
+				target, err := os.Readlink(mountPath)
```
Comment (tomponline) [inline lxd/storage/drivers/generic_vfs.go:562]: @monstermunchkin for consistency, should we add this same logic to the bit above where it has `filepath.Walk` around where it says "Copying custom config volume"?

Comment (monstermunchkin) [reply lxd/storage/drivers/generic_vfs.go:562]: Nope. That code is only executed `if v.IsVMBlock()`. The `"Copying custom config volume"` log message will actually never show up and could be removed safely.

Comment (tomponline) [reply lxd/storage/drivers/generic_vfs.go:562]: True, although won't this become relevant again once you integrate it with https://github.com/lxc/lxd/pull/9300?

Comment (monstermunchkin) [reply lxd/storage/drivers/generic_vfs.go:562]: No, it won't. The `if v.IsVMBlock()` is placed around the VM config volume in #9300.

Comment (tomponline) [reply lxd/storage/drivers/generic_vfs.go:562]: OK great thanks.

**Review by tomponline — APPROVED** [review]

**Pull request merged**